helm/external-dns-app/templates/_helpers.tpl

```diff
@@ -207,8 +207,13 @@ Set Giant Swarm podAnnotations.
 Set Giant Swarm serviceAccountAnnotations.
 */}}
 {{- define "giantswarm.serviceAccountAnnotations" -}}
-{{- if and (or (eq .Values.provider "aws") (eq .Values.provider "capa")) (eq .Values.aws.irsa "true") (eq .Values.aws.access "internal") (not (hasKey .Values.serviceAccount.annotations "eks.amazonaws.com/role-arn")) }}
+{{- if and (eq .Values.aws.irsa "true") (eq .Values.aws.access "internal") (not (hasKey .Values.serviceAccount.annotations "eks.amazonaws.com/role-arn")) }}
+{{- if (eq .Values.provider "aws") }}
 {{- $_ := set .Values.serviceAccount.annotations "eks.amazonaws.com/role-arn" (tpl "arn:aws:iam::{{ .Values.aws.accountID }}:role/{{ template \"aws.iam.role\" . }}" .) }}
+{{- end}}
+{{- if and (or (eq .Values.provider "capa")) }}
+{{- $_ := set .Values.serviceAccount.annotations "eks.amazonaws.com/role-arn" (tpl "{{ template \"aws.iam.role\" . }}" .) }}
```

Comment:
```suggestion
{{- $_ := set .Values.serviceAccount.annotations "eks.amazonaws.com/role-arn" (include "aws.iam.role" .) }}                                                                                                                                        
```

nitpicking here but the `tpl` + `template` is not needed anymore.

Reply:
👊🏾